common/djangoapps/student/roles.py

```diff
@@ -310,3 +311,24 @@ def courses_with_role(self):
         * role (will be self.role--thus uninteresting)
         """
         return CourseAccessRole.objects.filter(role=self.role, user=self.user)
+
+
+def get_aggregate_exclusion_user_ids(course_key):
+    """
+    This helper method will return the list of user ids that are marked in roles
+    that can be excluded from certain aggregate queries. The list of roles to exclude
+    can be defined in a AGGREGATION_EXCLUDE_ROLES settings variable
+    """
+
+    exclude_user_ids = set()
+    exclude_role_list = getattr(settings, 'AGGREGATION_EXCLUDE_ROLES', [CourseObserverRole.ROLE])
+
+    for role in exclude_role_list:
```

Comment:
Is it possible to avoid doing N queries here using the Django ORM?  I feel like it's possible to write some SQL to generate the set of excluded users in a single query, so it should be possible to do it via Django as well.